src/compton/test/t2Compton.cc

```diff
@@ -0,0 +1,322 @@
+    compton_test.reset(new rtt_compton::Compton(filename));
+  } catch (int /*asrt*/) {
+    FAILMSG("Failed to construct a Compton object!");
+    // if construction fails, there is no reason to continue testing...
+    return;
+  }
+  std::cout << "\n(...Success!)" << std::endl;
+
+  // Check some of the data in the CSK_generator-opened file:
+  const std::vector<double> grp_bds = compton_test->get_group_bounds();
+  const std::vector<double> T_evals = compton_test->get_etemp_pts();
+
+  // Unitless (divided by mec2)
+  // NB: These values can be read directly from the ASCII data files
+  // (3rd line)
+  std::vector<double> grp_bds_gold = {1.57311251e-06, 3.14622503e-04, 7.86556258e-04,
+                                      1.57311251e-03, 3.14622503e-02};
+  // (scattered throughout data file)
+  std::vector<double> T_evals_gold = {1.57311251e-05, 1.57311251e-04, 3.30353629e-04,
+                                      6.60707256e-04};
+  // First and last temperature from line 2 of the csk input file
+  std::vector<double> line2_Ts_gold = {1.41580126e-05, 7.26777982e-04};
+
+  // Sizes
+  const size_t num_groups_gold = grp_bds_gold.size() - 1U;
+  const size_t num_T_evals_gold = T_evals_gold.size();
+  const size_t num_evals_gold = 4; // out_lin, in_lin, out_nonlin, in_nonlin
+  const size_t num_leg_moments_gold = 2;
+
+  // Multiply by electron rest-mass energy (keV; using CSK value)
+  const double mec2 = 510.998;
+  for (size_t i = 0; i < grp_bds_gold.size(); ++i) {
+    grp_bds_gold[i] *= mec2;
+  }
+  // Interface does not scale temperatures, so no multiplication needed:
+  /*
+  for (size_t i = 0; i < T_evals_gold.size(); ++i)
+  {
+      T_evals_gold[i] *= mec2;
+  }
+  line2_Ts_gold[0] *= mec2;
+  line2_Ts_gold[1] *= mec2;
+  */
+
+  Ensure(grp_bds.size() == (num_groups_gold + 1U));
+  ut.check(std::equal(grp_bds.begin(), grp_bds.end(), grp_bds_gold.begin(),
+                      [tol](double a, double b) -> bool { return soft_equiv(a, b, tol); }),
+           "checked group boundaries");
+
+  Ensure(T_evals.size() == num_T_evals_gold);
+  ut.check(std::equal(T_evals.begin(), T_evals.end(), T_evals_gold.begin(),
+                      [tol](double a, double b) -> bool { return soft_equiv(a, b, tol); }),
+           "checked temperature grid");
+
+  if (!soft_equiv(compton_test->get_min_etemp(), line2_Ts_gold[0]))
+    FAILMSG("Min etemp read incorrectly!");
+  if (!soft_equiv(compton_test->get_max_etemp(), line2_Ts_gold[1]))
+    FAILMSG("Max etemp read incorrectly!");
+
+  if (ut.numFails == 0) {
+    std::cout << "\nCorrectly read group bounds and electron temps!" << std::endl;
+  }
+
+  // Test data retrieval: interpolate to a grid point in temperature
+  {
+    const double interp_T = T_evals[num_T_evals_gold - 1U];
+    std::cout << "Testing interpolation at T = " << (interp_T * mec2) << " keV\n";
+
+    std::vector<std::vector<std::vector<std::vector<double>>>> interp_data =
+        compton_test->interpolate_csk(interp_T);
```

Comment:
At some point, we should rethink this storage type.  This creates a lot of indirection.  (no change required for this PR.)

Reply:
Oh yeah. This exact point is a major reason why we're moving from compton (which uses vector<vector<vector<vector<double>>>>) to compton2 (which uses vector<double>)